keps/sig-api-machinery/4008-crd-ratcheting/README.md

```diff
@@ -0,0 +1,1272 @@
+
+#### Correlation of Old and New
+
+For comparison of `old`/`new` `maps`/`lists`/`sets`, we need a method for 
+correlation of the old and new elements. Whether that is normalizing the fields,
+creating a map representation of the object, or otherwise. This may be tricky
+to optimize, so care should be taken to meet performance goals.
+
+### Cel-Validator changes
+
+[`cel.Validate`](https://github.com/kubernetes/kubernetes/blob/2a50ef677ebf0729f176e852d0ad16950e122f6e/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go#L148) is already written to validate an update, 
+correlating old and new fields, so would only need to be modified to return an 
+ignored error list when applicable.
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+###### Schema-Changing Tests May Need Refactoring
+
+This change could cause some previously valid tests of invalid schema changes to
+fail. Those tests will have to be updated by changing to use create, or by
+changing the invalid fields during an update.
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `apiextensions-apiserver/pkg/apiserver/schema/cel`: `<date>` - `<test coverage>`
+- `apiextensions-apiserver/pkg/apiserver/schema/listtype`: `<date>` - `<test coverage>`
+- `apiextensions-apiserver/pkg/apiserver/schema/objectmeta`: `<date>` - `<test coverage>`
```

Comment:
not sure this is supposed to be exhaustive. We need heavy tests in kube-openapi.

Reply:
I've made a best effort to enumerate the packages for the implementation, and added their current test coverage